### PR TITLE
 system controller

### PR DESCRIPTION
**Pull request by psantos1113**
Convert SystemActions to use ActionResponse and convert tests to parameterized tests.

### CODE REVIEW

**Review by jamesrichard91 — APPROVED** [review]

**Review by gkillough — APPROVED** [review]

File: src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java [hunk 1]
```diff
@@ -1,157 +1,88 @@
...
-        systemActions.getSystemMessagesBetween("2018-11-13T00:00:00.000Z", "2018-11-13T01:00:00.000Z");
-        Mockito.verify(defaultSystemMessageUtility).findBetween(Mockito.any());
-    }
-
-    @Test
-    public void testGetSystemMessages() {
-        SystemActions systemActions = new SystemActions(defaultSystemStatusUtility, defaultSystemMessageUtility, fieldModelProcessor, settingsUtility);
-        systemActions.getSystemMessages();
-        Mockito.verify(defaultSystemMessageUtility).getSystemMessages();
+    @ParameterizedTest
```
Comment (bamandel) [inline src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java:51]: Are we sure we want to use a parameterized test here? It looks like we just want to create a start and end time list and iterate over it in the test. Parameterized tests will just make it harder to debug the test in my opinion.

**Review by bamandel — COMMENTED** [review]
I think parameterized tests should be used sparingly. If it errors out, it's hard to tell which test failed and it's hard to debug the specific failed test. When we had separate tests, we knew the specific test that failed and can easily debug it.

File: src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java (hunk 1 above)
Comment (jamesrichard91) [reply src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java:51]: This is what parameterized tests are meant for. I don't think we should change it

Comment (psantos1113) [reply src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java:51]: Parameterized tests are expanded into individual tests where the input is shown as part of the test.

Comment (bamandel) [reply src/test/java/com/synopsys/integration/alert/web/actions/SystemActionsTest.java:51]: Parameterized are meant for stuff like this, but ends up making things harder in my opinion. Why use a parameterized test when you can just pass the list yourself?

**Review by bamandel — APPROVED** [review]

**Review by ChomickiM — APPROVED** [review]

**Pull request merged**